xdsl/irdl/irdl.py

```diff
@@ -1581,6 +1604,20 @@ def get_constraint(
                         type_var_mapping=type_var_mapping,
                     )
 
+                # Get attribute constraints from a list of pyrdl constraints
+                def get_range_constraint(
+                    pyrdl_constr: (
+                        RangeConstraint
+                        | AttrConstraint
+                        | Attribute
+                        | type[Attribute]
+                        | TypeVar
+                    ),
+                ) -> RangeConstraint:
+                    if isinstance(pyrdl_constr, RangeConstraint):
+                        return pyrdl_constr
+                    return RangeOf(get_constraint(pyrdl_constr))
+
```

Comment:
I'm not sure I would define a new function if there is only a single use? Unless you expect it to have more uses?

Reply:
Yep, next PR makes it three uses :wink: 
Happy to move it to the next PR if it comes to that

Reply:
Assuming this is okay as it is an approval's comment, so merging as is; yell at me if I was wrong ^^